llm/server.go

```diff
@@ -200,6 +200,26 @@ func NewLlamaServer(gpus gpu.GpuInfoList, model string, ggml *GGML, adapters, pr
 		params = append(params, "--numa")
 	}
 
+	// Only enable flash_attn if all GPUs support it (CUDA 7+ or Metal)
+	flashAttnSupported := false
```

Comment:
I think here we want to check if `opts.NumGpu` is equal to the total number of layers earlier in this function (i.e. the model is fully offloaded to GPU)

Reply:
How about:

```
if uint64(opts.NumGPU) != ggml.KV().BlockCount() + 1 
```

Reply:
Looks good. I think we can clean it up a bit to something like this:

```
flashAttnSupported := true

// partial offloading does not support flash attention
if opts.NumGPU < ggml.KV().BlockCount() + 1 {
  flashAttnSupported = false
}

// only cuda (compute capability 7+) and metal support flash attention
for _, g := range gpus {
  if g.Library != "metal" && (g.Library != "cuda" || g.DriverMajor < 7) {
    flashAttnSupported = false
  }
}
```

Reply:
Done!

I'm quite new to Go (obviously) so I appreciate you steering me in the preferred direction.